### PR TITLE
Support for openssl 1.1.x in Swift-JWK-to-PEM

### PR DESCRIPTION
**Pull request by nethraravindran**

### CODE REVIEW

File: Sources/OpenSSL/shim.h [hunk 1]
```diff
@@ -196,4 +196,27 @@ static inline void EVP_CIPHER_CTX_free_wrapper(EVP_CIPHER_CTX *ctx) {
         #endif
 }
 
+// This wrapper allows for a common call for both versions of OpenSSL when setting other keys for RSA.
+static inline void RSA_set_keys(RSA *rsakey, BIGNUM *n, BIGNUM *e, BIGNUM *d, BIGNUM *p, BIGNUM *q, BIGNUM *dmp1, BIGNUM *dmq1, BIGNUM *iqmp) {
+
+	#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
```
Comment (ianpartridge) [inline Sources/OpenSSL/shim.h]: Please use `OPENSSL_VERSION_NUMBER >= 0x10100000L` to match the rest of the file.

File: Sources/OpenSSL/shim.h [hunk 2]
```diff
@@ -196,4 +196,27 @@ static inline void EVP_CIPHER_CTX_free_wrapper(EVP_CIPHER_CTX *ctx) {
...
+	#else
+		RSA_set0_key(rsakey, n, e, d);
+		RSA_set0_factors(rsakey, p, q);
+		RSA_set0_crt_params(rsakey, dmp1, dmq1, iqmp);
+	#endif
+}
+
+static inline void EVP_PKEY_assign_wrapper(EVP_PKEY *pkey, RSA *rsakey) {
+
+	EVP_PKEY_assign(pkey, EVP_PKEY_RSA, rsakey);
```
Comment (ianpartridge) [inline Sources/OpenSSL/shim.h:220]: I couldn't find this method in the docs - I could only see `EVP_PKEY_assign_RSA()`. Can you link me to it?  Also, if it's the same on all versions of OpenSSL why do we need to wrap it?

File: Sources/OpenSSL/shim.h (hunk 1 above)
Comment (nethraravindran) [reply Sources/OpenSSL/shim.h]: Sure, Thanks @ianpartridge

File: Sources/OpenSSL/shim.h (hunk 2 above)
Comment (nethraravindran) [reply Sources/OpenSSL/shim.h:220]: Here `RSA` is an opaque structure in _openssl 1.1.x_ and where as not in _openssl1.0.x_
Hence to provide the switch between `Opaque Pointer` and `Typed Pointers` this wrapper is needed 

Example: https://github.com/ibm-cloud-security/Swift-JWK-to-PEM/blob/53d67623c7067dd6613676cbebc1a1dd333ce027/Sources/SwiftJWKtoPEM/RSAKey%2BJWK.swift#L123

**Review by ianpartridge — APPROVED** [review]

**Review by billabt — APPROVED** [review]
👍

**Pull request merged**